apps/cloud_inference/server_app/include/inference.h

```diff
@@ -340,5 +348,43 @@ class InferenceEngineHip:public InferenceEngine
     hipStream_t         hip_stream[MAX_NUM_GPU];
 
 };
+
+class InferenceEngineRocalHip:public InferenceEngineHip
+{
+public:
+    InferenceEngineRocalHip(int sock_, Arguments * args, const std::string clientName, InfComCommand * cmd, const std::string folderPath);
+    ~InferenceEngineRocalHip();
+    int run();
+
+protected:
+
+    virtual void workMasterInputQ();
+    virtual void workDeviceInputCopy(int gpu);
+    virtual void workDeviceProcess(int gpu);
+    virtual void workDeviceOutputCopy(int gpu);
+
+    MessageQueue<std::tuple<int,char *,int>> inputQ;
+    // scheduler device queues
+    MessageQueue<int>                    * queueDeviceTagQ[MAX_NUM_GPU];
+    MessageQueue<std::string>            * queueDeviceNameQ[MAX_NUM_GPU];
+
+    vx_context openvx_context[MAX_NUM_GPU];
```

Comment:
all these members are defined in base class. You don't need to redefine here except for rocAL